.github/workflows/test.yaml

```diff
@@ -0,0 +1,16 @@
+name: build
+
+on: [push, pull_request]
+
+jobs:
+  test_micm_api:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
```

Comment:
This is a thing we've added in all of our other PRs. The intent is to only run the tests once when you have a PR. Otherwise, the tests run both on pushes when a PR is active and because it is a PR. If there's a better way, I'd like to know but this is the best we can figure out for now.

Reply:
@K20shores @boulderdaze if the issue is that the tests are running on PRs twice after every pushed commit then I am pretty sure you are running into the exact issue I described above about `push` and PR `synchronize` events triggering at the same time.  Might be worth trying the `On:` changes I listed above instead of the `if:` statement to see if that fixes that particular issue or not.

Of course if it doesn't solve the double-trigger problem then I am happy to leave this code as-is.